### PR TITLE
More information about scooters, elecrtric bikes, POI's

### PR DESCRIPTION
**Pull request by Antiik91**
## Proposed Changes

  -

## Pull Request Check List

  - A reasonable set of unit tests is included
  - Console does not show new warnings/errors
  - Changes are documented or they are self explanatory
  - This pull request does not have any merge conflicts
  - All existing tests pass in CI build

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform
  - Check that the implementation matches the design, when such one is defined in a Jira issue
  - Merge the pull request

### CODE REVIEW

File: app/component/itinerary/NaviDestination.js [hunk 1]
```diff
@@ -28,7 +32,7 @@ function NaviDestination({ leg, focusToLeg }) {
     <div className="navileg-destination-details">
       {toIcon && <Icon img={toIcon} className="navi-destination-icon" />}
       <div>
-        {stop?.name}&nbsp;
+        {stop ? stop?.name : placeName}
         {stop?.code && <StopCode code={stop.code} />}
```
Comment (vesameskanen) [inline app/component/itinerary/NaviDestination.js:36]: ? not needed, stop is already known to exist. stop.name || placeName

File: app/component/itinerary/NaviLeg.js [hunk 2]
```diff
@@ -14,17 +14,24 @@ const iconMap = {
...
+  let goTo = `navileg-${leg.mode.toLowerCase()}`;
 
+  if (isRental(leg, nextLeg)) {
+    if (leg.mode === 'WALK' && nextLeg?.mode === 'SCOOTER') {
+      goTo = `navileg-rent-scooter`;
+    } else {
+      goTo = `navileg-rent-cycle`;
+    }
+  }
   return (
```
Comment (vesameskanen) [inline app/component/itinerary/NaviLeg.js:28]: This is somehow against logic of other instructions. Anyway, let's see how this fits into the new design.

File: app/component/itinerary/Navigator.js [hunk 3]
```diff
@@ -51,9 +51,13 @@ function Navigator({ itinerary, focusToLeg, setNavigation }, context) {
     if (newLeg && newLeg !== currentLeg) {
       setCurrentLeg(newLeg);
       focusToLeg(newLeg, false);
+      // set nextLeg
+      const next = itinerary.legs.find(
+        leg => legTime(leg.start) > legTime(newLeg.start),
+      );
+      setNextLeg(next);
     }
```
Comment (vesameskanen) [inline app/component/itinerary/Navigator.js:54]: I wonder if next leg could simply be searched just before rendering NaviLeg. It does not have to be  in component state.

File: app/util/legUtils.js [hunk 4]
```diff
@@ -606,6 +606,18 @@ export const showBikeBoardingNote = (leg, config) => {
   );
 };
 
+/**
+ * Determines whether to current or next leg  contains  rental vehicles
+ * @param {object} leg - The leg object.
```
Comment (vesameskanen) [inline app/util/legUtils.js:611]: Doc is not true - current leg is WALK.

File: app/translations.js [hunk 5]
```diff
@@ -1269,16 +1269,19 @@ const translations = {
...
     'modes.to-bus': 'bus stop',
     'modes.to-car-park': 'car park',
     'modes.to-place': 'destination',
     'modes.to-rail': 'train station',
+    'modes.to-scooter': 'to scooter',
     'modes.to-stop': 'stop',
     'modes.to-subway': 'subway station',
     'modes.to-tram': 'tram stop',
+    'mods.to-scooter': 'scooter',
     'more-departures': 'More departures',
```
Comment (vesameskanen) [inline app/translations.js]: remove mods

**Review by vesameskanen — CHANGES_REQUESTED** [review]

File: app/component/itinerary/NaviDestination.js (hunk 1 above)
Comment (Antiik91) [reply app/component/itinerary/NaviDestination.js:36]: Stop can be null, for example when first leg is walk to a scooter. In these cases, we will experience crash without` ? `

**Pull request merged**